lib/partitionMessages.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright 2014 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+"use strict";
+
+var utils = require('./utils');
+
+/**
+ * PartitionMessages resource, scoped to a single Partition.
+ * @type {Function}
+ */
+var PartitionMessages = module.exports = function(partition) {
+    this.client = partition.client;
+    this.partition = partition;
+}
+
+/**
+ * Request a specified number of messages from a specified offset
+ *
+ * type options {
+ *      offset: number,
+ *      count: ?number,
+ *      format: ?string (json, avro, binary)
+ *  }
+ *
+ * @param options
+ * @param res function(err, messages)
+ */
+PartitionMessages.prototype.messages = function(options, res) {
+    options = options || {};
+
+    // Ensure that offset is passed in
+    if (typeof options === 'function' || options.offset === undefined || options.offset === null) {
```

Comment:
Ah, that makes sense. I guess this just protects the user from a more confusing error happening later since an offset is always required anyway? Doesn't hurt to do this validation and make it clearer to the user why their call is failing.